### PR TITLE
Propagate S3 errors through NSError object userInfo["Errors"]

### PR DESCRIPTION
**Pull request by k4ety**
- This change captures any response from S3 when uploading using AWSS3TransferUtility (any response was previously ignored)
- For AWSS3TransferUtility, any S3 response (either upload or download) that includes an "Errors" xml object logs the errors at the debug level
- For AWSS3TransferUtility, any S3 response (either upload or download) that includes an "Errors" xml object passes the errors through the completionHandler NSError object as error.userInfo["Errors"], so that user code has access to the S3 errors
- Fixed minor misspellings and warnings from Xcode

### CODE REVIEW

**Comment by kvasukib** [comment]
Hello Paul,

Thank you for using our SDK and submitting a pull request. Can you resolve the conflicts and resubmit it?

**Comment by k4ety** [comment]
Resolved conflicts

**Comment by kvasukib** [comment]
Hi Paul,

Thanks for resolving the conflicts. We will take a look at it.

**Comment by k4ety** [comment]
I resolved conflicts once more.  It would really be nice if someone would review the changes and go ahead with the merge.

**Comment by k4ety** [comment]
I have been waiting and keeping this up-to-date for five months and I see no process to request a review.  If anyone is able to review and merge it would be appreciated.  

As referenced above, I believe that this issue #757, as well as providing useful functionality, in general.

File: AWSLex/AWSLexModel.h [hunk 1]
```diff
@@ -117,9 +117,7 @@ typedef NS_ENUM(NSInteger, AWSLexDialogState) {
...
- */
+ // <p> You pass this value as the <code>Accept</code> HTTP header. </p><p> The message Amazon Lex returns in the response can be either text or speech based on the <code>Accept</code> HTTP header value in the request. </p><ul><li><p> If the value is <code>text/plain; charset=utf-8</code>, Amazon Lex returns text in the response. </p></li><li><p> If the value begins with <code>audio/</code>, Amazon Lex returns speech in the response. Amazon Lex uses Amazon Polly to generate the speech (using the configuration you specified in the <code>Accept</code> header). For example, if you specify <code>audio/mpeg</code> as the value, Amazon Lex returns speech in the MPEG format.</p><p>The following are the accepted values:</p><ul><li><p>audio/mpeg</p></li><li><p>audio/ogg</p></li><li><p>audio/pcm</p></li><li><p>text/plain; charset=utf-8</p></li><li><p>audio/* (defaults to mpeg)</p></li></ul></li></ul>
```
Comment (kvasukib) [inline AWSLex/AWSLexModel.h]: AWSLexModel.h is being auto-generated by the SDK Code Generator. Modifications to the auto-generated file will be overwritten by the output of codegen during every release. Can you revert just the change to AWSLexModel.h file?

File: AWSiOSSDKv2.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kvasukib) [inline AWSiOSSDKv2.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings]: What's the use of this file?

Comment (k4ety) [reply AWSiOSSDKv2.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings]: This was an inadvertent change and can be deleted.

**Comment by k4ety** [comment]
Reverted and removed.  Should be good to go.  
Thanks...

File: AWSLex/AWSLexModel.h (hunk 1 above)
Comment (k4ety) [reply AWSLex/AWSLexModel.h]: Reverted

**Comment by kvasukib** [comment]
Thanks @k4ety for the quick reply. I am verifying the changes.

**Comment by k4ety** [comment]
Any update?

File: AWSS3/AWSS3TransferUtility.m [hunk 3]
```diff
@@ -762,6 +776,30 @@ - (void)URLSession:(NSURLSession *)session
...
+        if ([responseString rangeOfString:@"<Error>"].location != NSNotFound) {
+          AWSXMLDictionaryParser *xmlParser = [AWSXMLDictionaryParser new];
+          xmlParser.trimWhiteSpace = YES;
+          xmlParser.stripEmptyNodes = NO;
+          xmlParser.wrapRootNode = YES; //wrapRootNode for easy process
+          xmlParser.nodeNameMode = AWSXMLDictionaryNodeNameModeNever; //do not need rootName anymore since rootNode is wrapped.
+          
+          NSDictionary *responseDict = [xmlParser dictionaryWithString:responseString];
+          userInfo[@"Error"] = responseDict[@"Error"];
+          AWSDDLogDebug(@"Error response received from S3: %@", responseDict);
```
Comment (rohandubal) [inline AWSS3/AWSS3TransferUtility.m]: Can you please change this to `AWSDDLogError`

Comment (k4ety) [reply AWSS3/AWSS3TransferUtility.m]: Done

**Comment by k4ety** [comment]
Instead of adding another test to AWSS3TransferUtilityTests I added example code to demonstrate handling error info from S3 with existing tests.  If this is not sufficient, please let me know what you think would be a good test.

**Comment by k4ety** [comment]
I have been trying for almost a year to get this pull request implemented.  If there is a problem with the request then someone should say so.  I am tired of going through the effort of resolving conflicts time after time just to have the the request ignored.  The remaining conflicts are trivial, but this seems pointless.

**Comment by k4ety** [comment]
The most recent changes to AWSTransferUtility accomplished 90% of my previous changes by capturing the responseData from each request making it available in its corresponding transfer utility task.  In the  current pull request I have add back the following:

- For AWSS3TransferUtility, any S3 response that includes an "Errors" xml object logs the errors at the debug level
- For AWSS3TransferUtility, any S3 response that includes an "Errors" xml object passes the errors through the completionHandler NSError object as error.userInfo["Errors"], so that user code has access to the S3 errors

**Comment by k4ety** [comment]
@rohandubal, please take a look and lets see if we can get this merged before more conflicts are introduced...

**Comment by rohandubal** [comment]
Hello @k4ety apologies for missing this PR. I am reviewing it currently.

**Pull request merged**